app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue

```diff
@@ -96,41 +95,134 @@ const widgetConfiguration = computed(() => {
+        props.onUpdate(argApp.appTree.function, argApp.appTree.exprId)
+        return true
+      } else if (value == null && argApp.appTree instanceof Ast.OprApp) {
+        /* Case: Removing infix application. */
+
+        // Infix application is removed as a whole. Only the target is kept.
+        if (argApp.appTree.lhs) {
+          props.onUpdate(argApp.appTree.lhs, argApp.appTree.exprId)
+        }
+        return true
+      } else if (argApp.appTree instanceof Ast.App && argApp.appTree.argumentName == null) {
+        /* Case: Removing positional prefix argument. */
+
+        // Since the update of this kind can affect following arguments, it is necessary to
+        // construct the new AST for the whole application in order to update it. Because we lack
+        // the ability to do AST manipulation directly, we are forced to do it by string operations.
+        // FIXME[#8367]: Edit application AST instead of concatenating strings.
+        let newRepr = ''
+
+        // Traverse the application chain, starting from the outermost application and going
+        // towards the innermost target.
+        for (let innerApp of app.iterApplications()) {
+          if (innerApp === argApp) {
+            // Found the application with the argument to remove. Skip the argument and use the
+            // application target's code. This is the final iteration of the loop.
+            newRepr = `${argApp.appTree.function.repr().trimEnd()} ${newRepr.trimStart()}`
+            // Perform the actual update, since we already have the whole new application code
+            // collected.
+            props.onUpdate(newRepr, app.appTree.exprId)
+            return true
+          } else {
+            // Process an argument to the right of the removed argument.
+            assert(innerApp.appTree instanceof Ast.App)
+            const argRepr = innerApp.appTree
+              .repr()
+              .substring(innerApp.appTree.function.repr().length)
+              .trim()
+            if (
+              innerApp.argument instanceof ArgumentAst &&
+              innerApp.appTree.argumentName == null &&
+              innerApp.argument.info != null
+            ) {
+              // Positional arguments following the deleted argument must all be rewritten to named.
+              newRepr = `${innerApp.argument.info.name}=${argRepr} ${newRepr.trimStart()}`
+            } else {
+              // All other arguments are copied as-is.
+              newRepr = `${argRepr} ${newRepr.trimStart()}`
+            }
+          }
+        }
+        assertUnreachable()
+      } else if (value == null && argApp.argument instanceof ArgumentPlaceholder) {
+        /* Case: Removing placeholder value. */
+        // Do nothing. The argument already doesn't exist, so there is nothing to update.
+        return true
+      } else {
+        // Any other case is handled by the default handler.
+        return false
+      }
+    }
+  }
+  return false
+}
```

Comment:
I plan to refactor this code by a lot once we have the AST based edits. Then it will be split into a separate testable logic. Before that there isn't much point in doing it.